src/main/java/org/mockito/quality/Strictness.java

```diff
@@ -89,5 +94,49 @@
      * @since 2.3.0
      */
     @Incubating
-    STRICT_STUBS;
+    STRICT_STUBS,
+
+    /**
+     * Enforces declaring stubbings for all <strong>non-void</strong> invocations.
+     * Please prefer {@link #STRICT_STUBS} for cleaner tests.
+     * If you are convinced you need {@code STRICT_MOCKS} mocking then read on.
+     * <p>
+     * This mode requires the developer to declare all stubbings for <strong>non-void</strong> methods
+     * invoked in the production code.
+     * Please remember that <strong>void methods</strong> are ignored and you are not required to stub them.
+     * <p>
+     * {@code STRICT_MOCKS} mocking is similar to traditional mocking with EasyMock,
+     * where undeclared invocations would fail early.
+     * {@code STRICT_MOCKS} mocking is useful for users migrating from EasyMock
+     * who are emotionally attached to stricter mocking.
+     * Mockito team recommends using {@link #STRICT_STUBS} which gets you the best of both worlds:
+     * clean tests and improved debuggability.
+     * {@code STRICT_MOCKS} mocking may increase brittleness of your tests
+     * and force you to declare additional stubbings that are irrelevant in given test method.
+     * This makes tests bigger and hence harder to read and comprehend.
+     * At times, {@code STRICT_MOCKS} mocking can be useful for troubleshooting tests or production code
+     * that is legacy or overcomplicated.
+     * <p>
+     * For discussion about {@code STRICT_MOCKS} mocking see ticket <a href="https://github.com/mockito/mockito/issues/1097">#1097</a>.
+     * <p>
+     * You can enable {@code STRICT_MOCKS} mocking via our JUnit4 support ({@link MockitoJUnit}) or {@link MockitoSession}
```

Comment:
The JUnit4 support is annotated with `@Deprecated` in this PR. If we want to deprecate it, we should not mention it in this javadoc.

Reply:
I deprecated only one specific JUnit4 runner subclass, not the entire JUnit4 support.